alica_engine/src/engine/BehaviourPool.cpp

```diff
@@ -74,11 +74,26 @@ void BehaviourPool::stopAll()
             ALICA_ERROR_MSG("BP::stop(): Found Behaviour without an BasicBehaviour attached!");
             continue;
         }
-
         bbPtr->stop();
     }
 }
 
+/**
+ * Calls terminate on all BasicBehaviours.
+ */
+void BehaviourPool::terminateAll()
+{
+    const PlanRepository::Accessor<BehaviourConfiguration>& behaviourConfs = _ae->getPlanRepository()->getBehaviourConfigurations();
+    for (const BehaviourConfiguration* beh : behaviourConfs) {
```

Comment:
You can directly iterate over the _availableBehaviours map. Example:

for( auto& behaviour : _availableBehaviours  ) {
        behaviour.second->terminate();
 }